### PR TITLE
Updates to docs, removing unneeded flags, updating buckets

### PR DESCRIPTION
**Pull request by JDBraun**
This PR address the following issues:
- #64 
- #65 
- #62 
- #61 

Most notably, the external location and data bucket are no longer required given the fact that catalog isolation has been implemented.

### CODE REVIEW

File: aws/tf/sra.tf [hunk 1]
```diff
@@ -25,9 +25,12 @@ module "SRA" {
   // Account - Unity Catalog:
   metastore_id            = null // Metastore configuration - leave null if there is no existing regional metastore, does not create a root storage location
   metastore_name          = join("", [var.resource_prefix, "-", var.region, "-", "uc"])
-  data_bucket             = null // REQUIRED - Existing S3 bucket name (e.g. data-bucket-s3-test)
   workspace_catalog_admin = null // REQUIRED - Workspace specific catalogs are created, this user will become an admin of that catalog (e.g. firstname.lastname@company.com)
-  external_location_admin = null // REQUIRED - Read-only external location is created, this user will become an admin of that exteranl location (e.g. firstname.lastname@company.com)
+
+  // Account - Unity Catalog - Data:
+  enable_read_only_external_location = false // Creates a read-only external location and corresponding IAM role, based on the value of the data_bucket variable
+  read_only_data_bucket              = null  // Accepts an S3 path (e.g. "s3://data-bucket")
```
Comment (ryan-gord-db) [inline aws/tf/sra.tf:32]: The example in the comment references the bucket URL, but in file `uc_external_location.tf`, this variable has the `s3://` prefix prepended.  Just the bucket name works so _(e.g. "<del>s3://</del>data-bucket")_ --> _(e.g. "data-bucket")_

**Comment by JDBraun** [comment]
@ryan-gord-db - great catch, updated!

File: aws/tf/modules/sra/variables.tf [hunk 2]
```diff
@@ -82,6 +81,12 @@ variable "enable_cluster_boolean" {
   sensitive   = true
 }
 
+variable "enable_read_only_external_location" {
+  description = "Flag to enable read only external location"
```
Comment (metrocavich) [inline aws/tf/modules/sra/variables.tf:85]: You mentioned this is a read only, optional bucket, so shouldn't we add a default value of false here that way you don't have to define this variable to use the module?

Comment (metrocavich) [reply aws/tf/modules/sra/variables.tf:85]: I see its nullable so you can pass in null, but I think that's confusing.

Comment (metrocavich) [reply aws/tf/modules/sra/variables.tf:85]: Right, I just mean that the best practice for writing a module being to provide default values for "optional" fields


Right now you have to include a line that sets it to false, which is how we have it structured in the example, but I think best practice is to set default variables here to indicate a true "optional" field


Also to indicate to any readers that that generally we would expect it to be false.

**Comment by JDBraun** [comment]
@metrocavich - updated in new commit, let me know

I took this and applied to all the optional variables of enabling certain features

**Comment by metrocavich** [comment]
Left a react but forgot to comment, makes a lot more sense to me, thanks for humoring me.

**Comment by JDBraun** [comment]
@metrocavich - let me know if you have anything else or will get it merged

**Comment by metrocavich** [comment]
This looks all fantastic to me now.
Very nice work.

**Pull request merged**